### PR TITLE
[ROS2] port grid_map_sdf

### PR DESCRIPTION
**Pull request by SteveMacenski**
- linted with uncrustify / cpplint
- changed long -> int64_t
- changing short -> int16_t
- adding inline to statics to suppress warnings for unused functions
- commented out pcl_ros function and headers because its not ported to ROS2 yet
- usual bits in cmake / package.xml
- rename h to hpp

### CODE REVIEW

**Comment by SteveMacenski** [comment]
If this is OK, before merging I should open a ticket / comment in the ROS2 port that we need pcl_ros in ROS2 to finish this package

**Comment by SteveMacenski** [comment]
@marwan99 please take a look

**Comment by SteveMacenski** [comment]
Always fun when I see different things locally versus CI

File: grid_map_sdf/include/grid_map_sdf/distance_transform/pnmfile.h [hunk 1]
```diff
@@ -18,20 +18,23 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
...
-#include "image.h"
-#include "misc.h"
+#include "grid_map_sdf/distance_transform/image.h"
+#include "grid_map_sdf/distance_transform/misc.h"
 
 namespace distance_transform
 {
 
-  const short BUF_SIZE = 256;
+  const int16 BUF_SIZE = 256;
```
Comment (Marwan99) [inline grid_map_sdf/include/grid_map_sdf/distance_transform/pnmfile.h]: I can see it compiling fine, but I am not sure if other compiler will be happy without "_t" in `int16`.

**Comment by Marwan99** [comment]
> Always fun when I see different things locally versus CI

Gets more fun when things don't work locally but work in the CI :')

File: grid_map_sdf/include/grid_map_sdf/distance_transform/pnmfile.h (hunk 1 above)
Comment (SteveMacenski) [reply grid_map_sdf/include/grid_map_sdf/distance_transform/pnmfile.h]: will do, sorry that one was missed in my second run through

**Comment by SteveMacenski** [comment]
> Gets more fun when things don't work locally but work in the CI :')

That one's less usual to me haha

**Comment by SteveMacenski** [comment]
Aight - GTG, @Marwan99 take a look

File: grid_map_sdf/src/SignedDistanceField.cpp [hunk 2]
```diff
@@ -152,33 +167,33 @@ Vector3 SignedDistanceField::getDistanceGradientAt(const Position3& position) co
   j = std::max(j, 1);
   j = std::min(j, size_.y() - 2);
   k = std::max(k, 1);
-  k = std::min(k, (int)data_.size() - 2);
+  k = std::min(k, static_cast<int>(data_.size() - 2));
```
Comment (Marwan99) [inline grid_map_sdf/src/SignedDistanceField.cpp:170]:
```suggestion
  k = std::min(k, static_cast<int>(data_.size()) - 2);
```

**Pull request merged**